src/org/zaproxy/zap/extension/jxbrowserlinux32/ExtensionJxBrowserLinux32.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.jxbrowserlinux32;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.SwingUtilities;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.jxbrowser.ZapBrowserFrame;
+import org.zaproxy.zap.utils.DisplayUtils;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+public class ExtensionJxBrowserLinux32 extends ExtensionAdaptor {
+
+    public static final String NAME = "ExtensionJxBrowserLinux32";
+
+    private static final Logger LOGGER = Logger.getLogger(ExtensionJxBrowserLinux32.class);
+
+    private static final ImageIcon CHROMIUM_ICON = DisplayUtils.getScaledIcon(new ImageIcon(
+            ExtensionJxBrowserLinux32.class.getResource(ExtensionJxBrowserLinux32.RESOURCE + "/chromium.png")));
+
+    protected static final String RESOURCE = "/org/zaproxy/zap/extension/jxbrowserlinux32/resources";
+
+    private JButton launchBrowserButton;
+
+    public ExtensionJxBrowserLinux32() {
+        super(NAME);
+    }
+
+    @Override
+    public void init() {
+        super.init();
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        if (getView() != null) {
+
+            if (Constant.isLinux() && ! isOs64Bits()) {
+                // Only show if we're running on the right platform
+                View.getSingleton().addMainToolbarButton(this.getLaunchBrowserButton());

```

Comment:
While playing with the add-on I noticed that this button is not being removed when uninstalling the add-on (same for other extensions).